sanic_routing/router.py

```diff
@@ -367,51 +507,68 @@ def routes(self):
             [route for group in self.groups.values() for route in group]
         )
 
-    def optimize(self, src: t.List[Line]) -> None:
-        """
-        Insert NotFound exceptions to be able to bail as quick as possible,
-        and realign lines to proper indentation
-        """
-        offset = 0
-        current = 0
-        insert_at = set()
-        for num, line in enumerate(src):
-            if line.indent < current:
-                if not line.src.startswith("."):
-                    if offset < 0:
-                        offset += 1
-                    else:
-                        offset = 0
-
+    def _optimize(self, node) -> None:
+        warn(
+            "Router AST optimization is an experimental only feature. "
+            "Results may vary from unoptimized code."
+        )
+        if hasattr(node, "body"):
+            for child in node.body:
+                self._optimize(child)
+
+            # concatenate nested single if blocks
+            # EXAMPLE:
+            #      if parts[1] == "foo":
+            #          if num > 3:
+            # BECOMES:
+            #       if parts[1] == 'foo' and num > 3:
+            # Testing has shown that further recursion does not actually
+            # produce any faster results.
+            if self._is_lone_if(node) and self._is_lone_if(node.body[0]):
+                current = node.body[0]
+                nested = node.body[0].body[0]
+
+                values: t.List[t.Any] = []
+                for test in [current.test, nested.test]:
+                    if isinstance(test, ast.Compare):
+                        values.append(test)
+                    elif isinstance(test, ast.BoolOp) and isinstance(
+                        test.op, ast.And
+                    ):
+                        values.extend(test.values)
```

Comment:
Maybe add `else` just in case something else is found so that the test is not silently dropped?